editor/code_editor.cpp

```diff
@@ -1956,11 +1908,47 @@ void CodeTextEditor::remove_all_bookmarks() {
 	text_editor->clear_bookmarked_lines();
 }
 
+void CodeTextEditor::zoom_in() {
+	int s = text_editor->get_theme_font_size("font_size");
+	set_zoom_factor(zoom_factor * (s + MAX(1.0f, EDSCALE)) / s);
+}
+
+void CodeTextEditor::zoom_out() {
+	int s = text_editor->get_theme_font_size("font_size");
+	set_zoom_factor(zoom_factor * (s - MAX(1.0f, EDSCALE)) / s);
+}
+
+void CodeTextEditor::set_zoom_factor(float p_zoom_factor) {
+	int neutral_font_size = EDITOR_GET("interface/editor/code_font_size").operator int() * EDSCALE;
+	int s = round(CLAMP(p_zoom_factor, 0.25f, 3.0f) * neutral_font_size);
+	float new_zoom_factor = (float)s / neutral_font_size;
+
+	StringBuilder sb;
+	sb.append(itos(round(new_zoom_factor * 100)));
+	sb.append(" %");
+	zoom_button->set_text(sb.as_string());
+
+	if (text_editor->has_theme_font_size_override("font_size")) {
+		text_editor->remove_theme_font_size_override("font_size");
+	}
+	text_editor->add_theme_font_size_override("font_size", s);
+
+	if (new_zoom_factor != zoom_factor) {
+		zoom_factor = new_zoom_factor;
+		emit_signal(SNAME("zoomed"), new_zoom_factor);
```

Comment:
The signal causes `set_zoom_factor()` to be called in every CodeTextEditor, which emits the signal again and updates every script editor. So with e.g. 4 script tabs, you get 16 update calls. This should be sorted out, so that zoom change only updates every editor once.

Reply:
The shouldn't happen. There are multiple layers of redundancy checks to prevent this.

One tab detects interaction, sets new value, fires signal. That's 1 update. The plugin catches signal, then propagates the zoom to only the tabs that have different zoom. For 4 tabs, that's 3 updates. No additional updates will happen after this now that the zooms are all equal.

Reply:
Right, but the `ScriptEditor::_set_zoom_factor` is still called after every emission (which happens in every code editor when you change zoom) and it calls methods in every code editor. Even if they stop at the zoom comparison, that's still many unnecessary calls.

Reply:
> The only redundancy I can see here is that each time ScriptEditor catches a signal, it goes through a loop to check all the tabs. Is that what you mean by unnecessary calls?

Yes.